tools/base/fpgainfo-c/powerinfo.c

```diff
@@ -108,23 +113,28 @@ static fpga_result get_power_info(fpga_token tok, struct power_info *finfo)
 out:
 	return res;
 }
+#endif
 
-static void print_power_info(struct power_info *info)
+/*
+ * Print help
+ */
+void power_help(void)
 {
-	char guid_str[38];
-	uuid_unparse(info->guid, guid_str);
-	printf("//****** POWER ******//\n");
-	printf("%-24s : 0x%2lX\n", "Object Id", info->object_id);
-	printf("%-24s : 0x%02X\n", "Bus", info->bus);
-	printf("%-24s : 0x%02X\n", "Device", info->device);
-	printf("%-24s : 0x%02X\n", "Function", info->function);
-	printf("%-24s : 0x%02X\n", "Socket Id", info->socket_id);
-	printf("%-24s : %02d\n", "Ports Num", info->num_slots);
-	printf("%-24s : 0x%lX\n", "Bitstream Id", info->bbs_id);
-	printf("%-24s : 0x%lX\n", "Bitstream Metadata",
-	       *(uint64_t *)&info->bbs_version);
-	printf("%-24s : %s\n", "Pr Interface Id", guid_str);
-	// printf("%-24s : 0x%2lX\n", "Capabilities", info->capabilities);
+	printf("\nPrint power metrics\n"
+	       "        fpgainfo power [-h]\n"
+	       "                -h,--help           Print this help\n"
+	       "\n");
+}
+
+static void print_power_info(fpga_properties props)
+{
+	fpga_result res = FPGA_OK;
+
+	fpgainfo_print_common("//****** POWER ******//", props);
+
+	res = bmc_print_values(get_sysfs_path(props, FPGA_DEVICE, NULL),
```

Comment:
Will this only work on DCP platform?

Reply:
This reads IPMI-formatted binary from a sysfs file.  If other platforms provide it, then it will work.  But for now, it's A10 (Rush Creek) specific.